qualtran/bloqs/chemistry/thc/select_bloq.py

```diff
@@ -310,6 +307,39 @@ def build_composite_bloq(
             'sys_a': sys_a,
             'sys_b': sys_b,
         }
+        if self.control_val is not None:
+            out_soqs['control'] = soqs['control']
+
+        return out_soqs
+
+    def controlled(
+        self,
+        num_controls: Optional[int] = None,
+        control_values: Optional[
+            Union[cirq.ops.AbstractControlValues, Sequence[Union[int, Collection[int]]]]
+        ] = None,
+        control_qid_shape: Optional[Tuple[int, ...]] = None,
+    ) -> 'SelectTHC':
+        if num_controls is None:
+            num_controls = 1
+        if control_values is None:
+            control_values = [1] * num_controls
+        if (
+            isinstance(control_values, Sequence)
+            and isinstance(control_values[0], int)
+            and len(control_values) == 1
+            and self.control_val is None
+        ):
+            return SelectTHC(
+                num_mu=self.num_mu,
+                num_spin_orb=self.num_spin_orb,
+                num_bits_theta=self.num_bits_theta,
+                keep_bitsize=self.keep_bitsize,
+                control_val=control_values[0],
+            )
+        raise NotImplementedError(
+            f'Cannot create a controlled version of {self} with control_values={control_values}.'
+        )
```

Comment:
we should override `get_ctrl_system` instead of `controlled` for custom implementations: see https://github.com/quantumlib/Qualtran/issues/886

Reply:
Is there an example somewhere?

Reply:
from https://github.com/quantumlib/Qualtran/pull/904:

https://github.com/anurudhp/Qualtran/blob/308b60e891e0775b492ed70297ebbc929816c966/qualtran/bloqs/qubitization_walk_operator.py#L109-L133

Reply:
thanks!

Reply:
Done.